sections/git_branch.zsh

```diff
@@ -12,19 +12,53 @@ SPACESHIP_GIT_BRANCH_PREFIX="${SPACESHIP_GIT_BRANCH_PREFIX="$SPACESHIP_GIT_SYMBO
 SPACESHIP_GIT_BRANCH_SUFFIX="${SPACESHIP_GIT_BRANCH_SUFFIX=""}"
 SPACESHIP_GIT_BRANCH_COLOR="${SPACESHIP_GIT_BRANCH_COLOR="magenta"}"
 
+# Show current commit SHA configuration
+SPACESHIP_GIT_BRANCH_SHOW_COMMIT="${SPACESHIP_GIT_BRANCH_SHOW_COMMIT=false}"
+SPACESHIP_GIT_BRANCH_COMMIT_SYMBOL="${SPACESHIP_GIT_BRANCH_COMMIT_SYMBOL=" ¢ "}"
+SPACESHIP_GIT_BRANCH_PREFIX_COMMIT="${SPACESHIP_GIT_BRANCH_PREFIX_COMMIT=""}"
+SPACESHIP_GIT_BRANCH_SUFFIX_COMMIT="${SPACESHIP_GIT_BRANCH_SUFFIX_COMMIT=""}"
+
+# Show tag configuration
+SPACESHIP_GIT_BRANCH_SHOW_TAG="${SPACESHIP_GIT_BRANCH_SHOW_TAG=false}"
+SPACESHIP_GIT_BRANCH_TAG_SYMBOL="${SPACESHIP_GIT_BRANCH_TAG_SYMBOL=" ŧ "}"
+SPACESHIP_GIT_BRANCH_PREFIX_TAG="${SPACESHIP_GIT_BRANCH_PREFIX_TAG=""}"
+SPACESHIP_GIT_BRANCH_SUFFIX_TAG="${SPACESHIP_GIT_BRANCH_SUFFIX_TAG=""}"
+
 # ------------------------------------------------------------------------------
 # Section
 # ------------------------------------------------------------------------------
 
 spaceship_git_branch() {
   [[ $SPACESHIP_GIT_BRANCH_SHOW == false ]] && return
 
-  local git_current_branch="$vcs_info_msg_0_"
-  [[ -z "$git_current_branch" ]] && return
+  # Show commit/tag info on detached HEAD
+  if [[ $SPACESHIP_GIT_BRANCH_SHOW_COMMIT == true ]] || [[ $SPACESHIP_GIT_BRANCH_SHOW_TAG == true ]]; then
```

Comment:
I would suggest to compare with `"always"`, not with `true`, because the `else` branch actually also is capable of showing tags and commit hashes. This is also a pattern in other sections, a `SHOW` parameter that accepts `"always"`.